Jobs.Fetcher.YouTube/Helpers/ApiDataFetcher.cs

```diff
@@ -172,16 +177,21 @@ public static class ApiDataFetcher {
 
         public static List<(DateTime date, long subscriberViews)> FetchSubscriberViews(YTA.VideoDailyMetric mostRecentRecord, string channelId, YTD.Video video, DateTime now, YouTubeAnalyticsService analyticsService, Logger logger, bool reprocessMetrics = false) {
             DateTime mostRecentMetricDate;
-            if (mostRecentRecord == null || reprocessMetrics) {
-                TimeSpan PublishedAtOffset;
-                if (reprocessMetrics) {
-                    PublishedAtOffset = new TimeSpan(30, 0, 0, 0);
+            try {
+                if (mostRecentRecord == null || reprocessMetrics) {
+                    TimeSpan PublishedAtOffset;
+                    if (reprocessMetrics) {
+                        PublishedAtOffset = new TimeSpan(30, 0, 0, 0);
+                    } else {
+                        PublishedAtOffset = TimeMargin;
+                    }
+                    mostRecentMetricDate = (video.PublishedAt != null) ? video.PublishedAt - PublishedAtOffset : now;
                 } else {
-                    PublishedAtOffset = TimeMargin;
+                    mostRecentMetricDate = mostRecentRecord.Date;
                 }
-                mostRecentMetricDate = (video.PublishedAt != null) ? video.PublishedAt - PublishedAtOffset : now;
-            } else {
-                mostRecentMetricDate = mostRecentRecord.Date;
+            } catch (Exception e) {
+                logger.Error($"Error fetching most recent Metric for video {video.VideoId}");
+                throw e;
             }
 
```

Comment:
Hmmm So are there times in which the API sends empty rows and it is not null? And this might be what is causing the exception to be raised?

Reply:
I was getting this from Groups API. Empty (but not null) response, no errors... But then, I was making the wrong request. If this is generally working, maybe there is no reason to suspect something abnormal would happen

Reply:
I know that for some reason some videos do not send all the data requested. It might have something to do with a wrong request, but, like you said, it is generally working. I'll maybe change this to check if there are empty rows and make the warning, like you said. I'll update it right away